src/components/AddComment.jsx

```diff
@@ -0,0 +1,97 @@
+import React, { useState } from "react";
+import { postCommentByArticleId } from "../api/requests";
+
+import "./AddComment.css";
+
+const AddComment = ({ article_id, comments, setComments }) => {
+  const [comment, setComment] = useState("");
+  const [msg, setMsg] = useState("");
+  const [msgColor, setMsgColor] = useState("");
+  const [isPending, setIsPending] = useState(false);
+
+  const commentDisabled = isPending ? true : false;
+
+  const postNewComment = (e) => {
+    e.preventDefault();
+
+    const newCommentUI = {
+      body: comment,
+      votes: 0,
+      author: "weegembump",
+      article_id: article_id,
+      created_at: Date.now(),
+      comment_id: comments.length + 1,
+    };
+
+    setComments((currComments) => {
+      return [...currComments, newCommentUI];
+    });
+
+    setMsg("Adding comment...");
+    setIsPending(true);
+
+    postCommentByArticleId(article_id, {
+      username: "weegembump",
+      body: comment,
+    })
+      .then((res) => {
+        setMsgColor("green");
+        setMsg("Comment added");
+        setIsPending(false);
+        setTimeout(() => {
+          setMsg("");
+          setMsgColor("");
+        }, 3000);
+      })
+      .catch((err) => {
+        if (err) {
+          setMsgColor("red");
+          setMsg("Sorry, something went wrong");
+          setIsPending(false);
+          setTimeout(() => {
+            setMsg("");
+            setMsgColor("");
+          }, 3000);
+          setComments((currComments) => {
+            return currComments.map((comment, index) => {
+              if (comment.comment_id === newCommentUI.comment_id) {
+                currComments.splice(index, 1);
```

Comment:
`.splice` is a mutating method and shouldn't be used on `currComments`. Consider a non-mutating method like `.filter` to remove it instead